/*
*  This file is part of aasdk library project.
*  Copyright (C) 2018 f1x.studio (Michal Szwaj)
*
*  aasdk is free software: you can redistribute it and/or modify
*  it under the terms of the GNU General Public License as published by
*  the Free Software Foundation; either version 3 of the License, or
*  (at your option) any later version.

*  aasdk is distributed in the hope that it will be useful,
*  but WITHOUT ANY WARRANTY; without even the implied warranty of
*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
*  GNU General Public License for more details.
*
*  You should have received a copy of the GNU General Public License
*  along with aasdk. If not, see <http://www.gnu.org/licenses/>.
*/

syntax="proto2";

import "ChannelDescriptorData.proto";
import "ConnectionConfigurationData.proto";
import "HeadUnitInfoData.proto";

package oaa.proto.messages;

message ServiceDiscoveryResponse
{
    repeated data.ChannelDescriptor channels = 1;
    // Fields 2-11 are deprecated but kept for backward compatibility.
    // Modern phones read from headunit_info (field 17) instead.
    optional string head_unit_name = 2;
    optional string car_model = 3;
    optional string car_year = 4;
    optional string car_serial = 5;
    optional bool left_hand_drive_vehicle = 6;
    optional string headunit_manufacturer = 7;
    optional string headunit_model = 8;
    optional string sw_build = 9;
    optional string sw_version = 10;
    optional bool can_play_native_media_during_vr = 11;
    optional bool hide_clock = 12;

    // Modern fields (required by newer Android Auto versions)
    optional int32 session_configuration = 13;
    optional string display_name = 14;
    optional bool probe_for_support = 15;
    optional data.ConnectionConfiguration connection_configuration = 16;
    optional data.HeadUnitInfo headunit_info = 17;
}
